### ISSUE TITLE
Polling causes duplicate messages

### ISSUE THREAD

**Issue by MetRonnie** [cylc/cylc-flow#5526]
### Description

Polling a task that uses `cylc message` will result in those messages coming through again even if they have already been sent.

This is most obviously seen in the UI tree view.

### Expected Behaviour

(Need to define properly)

**Comment by wxtim** [comment]
I presume the the expected behaviour is not to get the message again - but is that correct @hjoliver , @oliver-sanders  - I can imagine scenarios where 2 copies of a message is more useful than 0 ?

**Comment by wxtim** [comment]
Very simply demonstrated with

```ini
[scheduling]
    initial cycle point = 1919
    [[graph]]
        R1 = foo & bar

[runtime]
    [[foo]]
        script = """
            for i in {1..5}; do
                echo $i
                cylc message -- "CRITICAL: [${i}] Hello World"
                sleep 5
            done
        """
    [[bar]]
        script = """
            for i in {1..5}; do
                echo $i
                cylc poll "${CYLC_WORKFLOW_ID}//${CYLC_TASK_CYCLE_POINT}/foo"
                sleep 5
            done
        """
```

```bash
cylc run --no-detach
```

Not completely sure it's a but - what if we are using the message to trigger outputs and we are polling because something appears to have gone wrong?

**Comment by oliver-sanders** [comment]
I can't think of a use case for logging the same message twice.

**Comment by wxtim** [comment]
What happens if Cylc message has failed on the remote and we are using poll as a backup? We might then get no messages - or could we put a switch in for the poll-as-backup scenario?

**Comment by oliver-sanders** [comment]
> What happens if Cylc message has failed on the remote and we are using poll as a backup?

All messages **should** get logged **once**, whether received via SSH/TCP or polling.

**Comment by hjoliver** [comment]
Yes, once per message regardless of comms method is the desired behavior.

Note, on current master:
- duplicate polled output messages do not re-trigger the downstream task (which wouldn't work anyway because of the "already spawned in this flow" check)
- but duplicate polled messages do re-run associated event handlers, which is not desirable

**Comment by wxtim** [comment]
So I can work out how to avoid logging duplicate polling messages (Check the new polled message against the activity log). But it's less clear at the moment how to stop the first polled message - 

If the polled line from `job.status` indicates `CYLC_MESSAGE` we know that the message has been sent. Can we assume that it has been received?

**Comment by wxtim** [comment]
Additional questions from a conversation with @dpmatthews :

> Why does the job activity log appear to show two separate log events for each poll?
>
> Can you get the original email to send you multiple task event handler emails? If not why?

**Comment by hjoliver** [comment]
@cylc/core - I had forgotten about this, but I think we should take another look at it soon - I'll put it to 8.7, optimistically - running event handlers many times for the same event is potentially quite bad. At best it's confusing to users who see this.

**Comment by oliver-sanders** [comment]
From our previous attempt to close this issue, it seems that it is very tricky, especially as there are multiple use patterns (both internal and external) running through the same interface.

I suspect we may need a refactor around this code to simplify the logic.
